btx/interfaces/istream.py

```diff
@@ -612,6 +640,20 @@ def launch_stream_analysis(in_stream, out_stream, fig_dir, tmp_exe, queue, ncore
     js.clean_up()
     js.submit()
 
+def get_most_recent_run(streams: list) -> int:
+    """ From a list of stream files, get the most recent run.
+
+    @param streams (list[str]) List of full paths to stream files.
+    @return run (int) Most recent run in the list of streams.
+    """
+    run: int = 0
+    for streampath in streams:
+        filename: str = streampath.split('/')[-1]
+        current_run = int(filename[1:5])
+        if current_run > run:
+            run = current_run
+    return run
```

Comment:
I agree with your point that this should be taken somewhere else eventually. 
Quoting for context:

> Functions created to help in stream_analysis and merge are non-general. Specifically, get_most_recent_run and get_most_recent_summary should be improved and moved to somewhere else in a refactor. They accomplish almost the exact same functionality but in a non-graceful and non-general way.